quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -543,4 +551,40 @@ private static boolean isRuntimeOption(String arg) {
 
         return arg.startsWith(ImportRealmMixin.IMPORT_REALM);
     }
+
+    private static void checkChangesInBuildOptionsDuringAutoBuild() {
+        if (Configuration.isOptimized()) {
+            List<PropertyMapper> buildOptions = stream(Configuration.getPropertyNames(true).spliterator(), false)
+                    .sorted()
+                    .map(PropertyMappers::getMapper)
+                    .filter(Objects::nonNull).collect(Collectors.toList());
+
+            if (buildOptions.isEmpty()) {
+                return;
+            }
+
+            StringBuilder output = new StringBuilder("You run a 'build' before starting the server, the following build options will be overridden to new values. " +
+                    "To avoid that, run the 'build' command again and start the server using the '").append(OPTIMIZED_BUILD_OPTION_LONG).append("' flag:");
+
+            for (PropertyMapper mapper : buildOptions) {
+                String newValue = ofNullable(getCurrentBuiltTimeProperty(mapper.getFrom()))
+                        .map(ConfigValue::getValue)
+                        .orElse("<unset>");
+                String currentValue = getRawPersistedProperty(mapper.getFrom()).get();
+
+                if (newValue.equals(currentValue)) {
+                    continue;
```

Comment:
What if all `newValue` equals to `currentValue`? :) Then we print `You run a 'build' before starting the server, the following build options will be overridden to new values.` with empty list below it?